Add Year End Poll and Modern Rock Madness contest schemas (#94)
ynotradio/site: docs/sanity-migration/11-neon-integration-modern-rock-madness.md

- **Performance**: PostgreSQL handles vote aggregation efficiently
- **Real-time updates**: Fast vote counting for live bracket updates
- **Scalability**: Database can handle high traffic during voting periods

### Data Flow

```
┌─────────────┐         ┌──────────────┐         ┌─────────────┐
│   Sanity    │◄────────│  Next.js API │────────►│    Neon     │
│             │         │              │         │  PostgreSQL │
│ • Tournament│         │ • Validate   │         │             │
│   Config    │         │ • Authorize  │         │ • Votes     │
│ • Groups     │         │ • Process    │         │ • Users     │
│ • Matches   │         │              │         │             │
└─────────────┘         └──────────────┘         └─────────────┘
```

1. User visits bracket → Frontend fetches tournament/match data from Sanity
2. User votes in matchup → API validates match is active
3. API stores vote in Neon PostgreSQL
4. Vote counts aggregated from Neon in real-time
5. Winner determined and updated in Sanity after match closes

---

## Sanity Schemas

### `modernRockMadnessTournament`
Annual tournament configuration.

> **Document ID format:** `modern-rock-madness-{year}` (e.g., `modern-rock-madness-2025`)

Reviewer: ```suggestion
**Document ID format:** `modern-rock-madness-{year}-{month}-{date}` (e.g., `modern-rock-madness-2025-03-31`)
```

**Key fields:**
- `title`: Display title (e.g., "Modern Rock Madness 2025")
- `year`: Tournament year
- `status`: `draft` | `active` | `complete` | `archived`
- `rounds`: Array of round configurations
  - `roundNumber`: Round sequence (1, 2, 3...)
  - `name`: Round name (e.g., "First Round", "Sweet Sixteen", "Elite Eight", "Final Four", "Championship")
  - `startsAt`: Round start datetime
  - `endsAt`: Round end datetime

### `modernRockMadnessGroup`
Tournament participant (group/artist). A group can represent one or more artists.

**Key fields:**
- `tournament`: Reference to `modernRockMadnessTournament` document
- `artists`: Array of references to `artist` documents (one or more artists)
- `seed`: Tournament seed (1-64 typical)
- `region`: Bracket region (e.g., "East", "West", "North", "South")
- `placement`: Final result (e.g., "Champion", "Runner-up", "Elite Eight")
- `sponsor`: Sponsor name for this group
- `abbreviation`: Short abbreviation for display

### `modernRockMadnessMatch`
Individual bracket matchup between two groups.

> **Document ID format:** `modern-rock-madness-{year}-match-{matchNumber}` (e.g., `modern-rock-madness-2025-match-1`)

Reviewer: ```suggestion
**Document ID format:** `modern-rock-madness-{year}-{month}-{date}-match-{matchNumber}` (e.g., `modern-rock-madness-2025-03-31-match-1`)
```

**Key fields:**
- `tournament`: Reference to `modernRockMadnessTournament` document
- `matchNumber`: Unique match number within tournament
- `round`: Round number (1 = First Round, 2 = Second Round, etc.)
- `region`: Bracket region
- `group1`: Reference to first `modern-rock-madnessGroup`
- `group2`: Reference to second `modern-rock-madnessGroup`

> - `winner`: Reference to winning `modern-rock-madnessGroup` (set after voting closes)

Reviewer: ```suggestion
- `group1`: Reference to first `modernRockMadnessGroup`
- `group2`: Reference to second `modernRockMadnessGroup`
- `winner`: Reference to winning `modernRockMadnessGroup` (set after voting closes)
```

The Modern Rock Madness tournament uses the same `votes` table as other contests, with these specific fields:

| Column | Type | Description |
|--------|------|-------------|
| `match_sanity_id` | VARCHAR(255) | Reference to `modernRockMadnessMatch` document |
| `option_sanity_id` | VARCHAR(255) | Reference to `modern-rock-madnessGroup` document (the group being voted for) |
| `user_id` | UUID | User who cast the vote |

**Unique constraint:** One vote per user per match (enforced via unique index on `user_id` and `match_sanity_id`)

**Note:** Write-ins are NOT supported for Modern Rock Madness (voters must choose one of the two groups in the matchup)

### Helper Functions

#### `get_modern_rock_madness_match_votes(p_match_id VARCHAR)`
Returns vote counts for a specific match.

**Returns:** `option_sanity_id` (group ID), `total_votes`

**Usage:**
```sql
SELECT * FROM get_modern_rock_madness_match_votes('modern-rock-madness-2025-match-1');
```

**Example result:**
```
option_sanity_id   | total_votes
-------------------+------------
mrmgroup-foo-2025  | 1523
mrmgroup-bar-2025  | 1204
```

---

## API Patterns

### Submitting a Vote

```typescript
// Example: Vote in a match

> POST /api/contests/mrm/vote

Reviewer: modern_rock_madness

  "winner": null,  // Set after match closes
  "votingActive": true
}
```

### Getting Bracket Status

```typescript
// Example: Get full tournament bracket
GET /api/contests/modern-rock-madness/bracket?tournament=modern-rock-madness-2025

Response:
{
  "tournamentId": "modern-rock-madness-2025",
  "year": 2025,
  "status": "active",
  "currentRound": 2,
  "regions": [
    {
      "name": "East",
      "matches": [...],
      "groups": [...]
    },
    ...
  ]
}
```

---

## Migration from Legacy MySQL

### Legacy Tables
- `modern-rock-madness_groups` - Tournament participants
- `modern-rock-madness_matches` - Bracket matchups
- `modern-rock-madness_votes` - Individual votes

### Migration Strategy
1. Create `modernRockMadnessTournament` document in Sanity for each year
2. Create `modern-rock-madnessGroup` documents for all tournament participants
3. Create `modernRockMadnessMatch` documents for all bracket matchups
4. Migrate votes from `modern-rock-madness_votes` to unified `votes` table
5. Update winner references in match documents based on vote counts

---

## Tournament Workflow

### Pre-Tournament Setup
1. Create `modernRockMadnessTournament` document with rounds configured

> 2. Add all `modern-rock-madnessGroup` documents with seeds and regions

Reviewer: ```suggestion
2. Add all `modernRockMadnessGroup` documents with seeds and regions
```
